small-talk/app/components/TopBar.test.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { render, fireEvent, screen } from '@testing-library/react';
+import TopBar from './TopBar';
+
+describe('TopBar Component', () => {
+  let originalLocation;
+  
+  beforeAll(() => {
+    originalLocation = window.location;
+    delete window.location;
+    window.location = { href: '', assign: jest.fn() };
+  });
+  
+  afterAll(() => {
+    window.location = originalLocation;
+  });
+
+  it('redirects to the correct path when the homepage button is clicked', () => {
+    render(<TopBar />);
+    const homeButton = screen.getByAltText('Logo');
+    fireEvent.click(homeButton);
+    expect(window.location.href).toBe('/homepage');
```

Comment:
Just tried adding a redirect test similar to yours in the SidebarButton test, and it didn't work. Would love to hear some feedback from you on that test's pull request (when I make it).